docs/source/user_guide/timeseries.ipynb

```diff
@@ -89,6 +90,159 @@
+    "\n",
+    "To show an example of this, let's create some discrepancies in our training data. We'll also add a couple of extra column in the X DataFrame to highlight more of the options with these tools."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X[\"Categorical\"] = [str(i%4) for i in range(len(X))]\n",
+    "X[\"Numeric\"] = [i for i in range(len(X))]\n",
+    "\n",
+    "# Re-split the data since we modified X\n",
+    "X_train, y_train = X.loc[train_dates], y.ww.loc[train_dates]\n",
+    "X_test, y_test =  X.loc[test_dates], y.ww.loc[test_dates]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_train[\"Date\"][500] = None\n",
+    "X_train[\"Date\"][1042] = None\n",
+    "X_train[\"Date\"][1043] = None\n",
+    "X_train[\"Date\"][231] = pd.Timestamp(\"1981-08-19\")\n",
+    "\n",
+    "X_train.drop(1209, inplace=True)\n",
+    "X_train.drop(398, inplace=True)\n",
+    "y_train.drop(1209, inplace=True)\n",
+    "y_train.drop(398, inplace=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With these changes, there are now NaN values in the training data that our models won't be able to recognize, and there is no longer a clear frequency between the dates."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Inferred frequency: {pd.infer_freq(X_train['Date'])}\")\n",
+    "print(f\"NaNs in date column: {X_train['Date'].isna().any()}\")\n",
+    "print(f\"NaNs in other training data columns: {X_train['Categorical'].isna().any() or X_train['Numeric'].isna().any()}\")\n",
+    "print(f\"NaNs in target data: {y_train.isna().any()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Time Series Regularizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the `TimeSeriesRegularizer` component to restore the missing and NaN DateTime values we've created in our data. This component is designed to infer the proper frequency using [Woodwork's `infer_frequency`](https://woodwork.alteryx.com/en/stable/generated/woodwork.statistics_utils.infer_frequency.html) function and generate a new DataFrame that follows it. In order to maintain as much original information from the input data as possible, all rows with completely correct times are ported over into this new DataFrame. If there are any rows that have the same timestamp as another, these will be dropped. The first occurrence of a data or time maintains priority. If there are any values that don\u2019t quite line up with the inferred frequency they will be shifted to any close missing times, or dropped if there are none nearby."
```

Comment:
The formatting of the link got messed up: 
![image](https://user-images.githubusercontent.com/41651716/164773351-2a0a6ebb-cf0d-49ec-9b7a-8f3551d2dac9.png)